mods/classic-taskbar-background-fix.wh.cpp

```diff
@@ -0,0 +1,368 @@
+// ==WindhawkMod==
+// @id              classic-taskbar-background-fix
+// @name            Classic Taskbar background fix
+// @description     Fixes Taskbar background in classic theme by replacing black background with a classic button face colour
+// @version         1.0
+// @author          Roland Pihlakas
+// @github          https://github.com/levitation
+// @homepage        https://www.simplify.ee/
+// @compilerOptions -luser32 -lgdi32
+// @include         explorer.exe
+// ==/WindhawkMod==
+
+// Source code is published under The GNU General Public License v3.0.
+//
+// For bug reports and feature requests, please open an issue here:
+// https://github.com/levitation-opensource/my-windhawk-mods/issues
+//
+// For pull requests, development takes place here:
+// https://github.com/levitation-opensource/my-windhawk-mods/
+
+// ==WindhawkModReadme==
+/*
+# Classic Taskbar background fix
+
+Fixes Taskbar background in classic theme by replacing black background with a classic button face colour.

```

Comment:
Can you add more details, e.g. which WIndows versions are supported? Which method of classic themes? A classic theme is not a native Windows feature anymore, right?

Reply:
I tested it both on Win10 and Win11. For classic theme I use Windhawk mod named classic-theme-enable. 

Classic theme still seems to be native to Windows. As I understand it, essentially this "classic-theme-enable" mod disables themes, and this reactivates the default behaviour of Windows, which is classic theme. 

There are a few registry parameters that need to be adjusted for nicer UI appearance. Also one might want to change colours. I use a program named "Desktop Themes" v1.87 for that. It is available here https://archive.org/details/desktop_themes_v187_ZIP

There are few programs that have minor visual glitches. Various Windhawk mods deal with this. 

Most problematic program is taskbar, which has a number of Windhawk mods, each fixing a different problem. Additionally I used ExplorerPatcher. The only problem that still manifested for me was this background issue. 

The only totally problematic program uncompatible with classic theme is Task Manager. I use Process Hacker instead. 

All other programs have been running fine. Right now my systems look entirely classic and I am quite intensive user using many different programs.

I guess I could write some document about my setup so that others could conveniently reproduce it.

Reply:
I assume Win11 is only supported with the Win10 taskbar, e.g. with ExplorerPatcher? Why not add it, and all the explanation that you've written here, even as-is, to the readme?

Some of the feedback I rightfully got for Windhawk is that it can be difficult to know which mods are relevant for which Windows versions. While I can add some tag to specify the relevant Windows versions, it's sometimes more complicated than that, e.g. with additional requirements such as having ExplorerPatcher installed. Windhawk is less helpful if this kind of information has to be deduced with trial and error or by asking in forums and chats.

Reply:
We definitely need tags, but more refined than just windows versions. Some suggestions: windws shell, cassic theme, gaming, taskbar, file manager...

Reply:
Tags are a different feature. Even if you have tags that say "taskbar" or "classic theme", it doesn't tell you what are the requirements, like Win10, Win11, ExplorerPatcher, etc.